File: app/src/main/java/io/edgedev/investment_calc/InvestmentDetail.java
package io.edgedev.investment_calc;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

public class InvestmentDetail extends Fragment {
    private static final String ARG_YEAR_KEY = "the_year";
    private MonthAdapter mMonthAdapter;
    private String year;


    public static InvestmentDetail newInstance(String year) {
        Bundle args = new Bundle();
        args.putString(ARG_YEAR_KEY, year);
        InvestmentDetail fragment = new InvestmentDetail();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        year = getArguments().getString(ARG_YEAR_KEY);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.activity_detail, container, false);

        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.monthRecycler);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        mMonthAdapter = new MonthAdapter();
        recyclerView.setAdapter(mMonthAdapter);

        List<Month> months = Singleton.getInstance().findMoney(year).getMonth();
        mMonthAdapter.addMonths(months);
        return view;
    }
}

File: app/src/main/java/io/edgedev/investment_calc/InvestmentDetail.kt
package io.edgedev.investment_calc

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class InvestmentDetail : androidx.fragment.app.Fragment() {
    private lateinit var mMonthAdapter: MonthAdapter
    private var year: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        year = arguments?.getString(ARG_YEAR_KEY)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val view = inflater.inflate(R.layout.activity_detail, container, false)

        val recyclerView = view.findViewById<View>(R.id.monthRecycler) as androidx.recyclerview.widget.RecyclerView
        recyclerView.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
        mMonthAdapter = MonthAdapter()
        recyclerView.adapter = mMonthAdapter

        val months : List<Month> = Singleton.getInstance().findMoney(year)!!.month
        mMonthAdapter.addMonths(months)
        return view
    }

    companion object {
        private val ARG_YEAR_KEY = "the_year"
        fun newInstance(year: String): InvestmentDetail {
            val args = Bundle()
            args.putString(ARG_YEAR_KEY, year)
            val fragment = InvestmentDetail()
            fragment.arguments = args
            return fragment
        }
    }
}